python/ray/data/tests/test_dataset.py

```diff
@@ -2532,142 +2519,6 @@ def udf(batch, a, b=None):
     values = [s["two"] for s in ds_list]
     assert values == [5, 7, 9]
```

Comment:
Shouldn't we still support constructor kwargs and such? I think the only thing we should remove is support de-ref of object references as args.

Reply:
Yep I'm adding support right now.

Reply:
@ericl Support added.